ambry-cloud/src/main/java/com.github.ambry.cloud/azure/CosmosDataAccessor.java

```diff
@@ -83,7 +83,7 @@ void testConnectivity() {
   public ResourceResponse<Document> upsertMetadata(CloudBlobMetadata blobMetadata) throws DocumentClientException {
     RequestOptions options = getRequestOptions(blobMetadata.getPartitionId());
     return retryOperationWithThrottling(
-        () -> documentClient.upsertDocument(cosmosCollectionLink, blobMetadata, options, true),
+        () -> documentClient.upsertDocument(cosmosCollectionLink, blobMetadata, null, true),
```

Comment:
this will require some discussion. There is problem with passing "options" here. I am getting this error: "com.microsoft.azure.documentdb.DocumentClientException: Message: {"Errors":["PartitionKey extracted from document doesn't match the one specified in the header"]}
". Not sure about the cause of this. Maybe @zzmao or @lightningrob can notice what I am missing here.

Reply:
We do need the options to contain the partition key.  We will need to debug the error cause and see what the two different values are.  Can you bring up in debugger and set a breakpoint there?